train_model.py

```diff
@@ -409,12 +409,11 @@ def get_params(self) -> dict:
             # Update parameters
             optimizer.step()
 
-            # Run validation args.validation_freq times per epoch. To do this,
-            # we split up the epoch into arg.validation_freq chunks and run
-            # validation after each chunk is finished.
-            progress_through_chunk = args.val_freq * (batch_idx + 1) \
-                                     / len(train_loader) % 1
-            if progress_through_chunk <= (args.val_freq-1) / len(train_loader):
+            # Run validation args.val_freq times per epoch. To do this, we split
+            # up the epoch into arg.validation_freq chunks and run validation
+            # after each chunk is finished.
+            if (num_val_runs + 1) / args.val_freq \
```

Comment:
If val_freq is set to 0, this will crash.

Reply:
Good catch! I realize that there were uninitialized variables that gave errors later on if validation was never run as well. I've changed it so that setting `--val-freq=0` is equivalent to saying you don't want any validation at all in [66f000c](https://github.com/DRAGNLabs/301r_retnet/pull/18/commits/66f000c460b548cd3e5a4f7072284d8bce3fc4a3).